File: app/src/main/java/com/yhjoo/dochef/adapter/MainAdPagerAdapter.java
package com.yhjoo.dochef.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import androidx.appcompat.widget.AppCompatImageView;
import androidx.viewpager.widget.PagerAdapter;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class MainAdPagerAdapter extends PagerAdapter {
        Context mContext;
        ArrayList<Integer> imgids;

        public MainAdPagerAdapter(Context context, ArrayList<Integer> imgids) {
            this.mContext = context;
            this.imgids = imgids;
        }

        @Override
        public Object instantiateItem(ViewGroup collection, int position) {
            AppCompatImageView aa = new AppCompatImageView(mContext);
            ViewGroup.LayoutParams lp = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
            aa.setLayoutParams(lp);
            Glide.with(mContext)
                    .load(imgids.get(position))
                    .centerInside()
                    .into(aa);

            collection.addView(aa);

            return aa;
        }

        @Override
        public void destroyItem(ViewGroup collection, int position, Object view) {
            collection.removeView((View) view);
        }

        @Override
        public int getCount() {
            return imgids.size();
        }

        @Override
        public boolean isViewFromObject(View view, Object object) {
            return view == object;
        }
    }

File: app/src/main/java/com/yhjoo/dochef/ui/adapter/MainAdPagerAdapter.kt
package com.yhjoo.dochef.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.widget.AppCompatImageView
import androidx.viewpager.widget.PagerAdapter
import com.bumptech.glide.Glide
import java.util.*

class MainAdPagerAdapter(var mContext: Context?, var imgids: ArrayList<Int>) : PagerAdapter() {
    override fun instantiateItem(collection: ViewGroup, position: Int): Any {
        val aa = AppCompatImageView(mContext!!)
        val lp = ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        aa.layoutParams = lp
        Glide.with(mContext!!)
            .load(imgids[position])
            .centerInside()
            .into(aa)
        collection.addView(aa)
        return aa
    }

    override fun destroyItem(collection: ViewGroup, position: Int, view: Any) {
        collection.removeView(view as View)
    }

    override fun getCount(): Int {
        return imgids.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }
}